components/core/CssSizeLength.cs

```diff
@@ -145,5 +139,66 @@ public static bool TryParse(string value, out CssSizeLength cssSizeLength)
             cssSizeLength = new(value);
             return cssSizeLength._parsed;
         }
+
+        private static bool TryResolveByPrefix(string trimmedValue, out CssSizeLengthUnit unit)
+        {
+            if (trimmedValue.StartsWith("calc", StringComparison.OrdinalIgnoreCase))
+            {
+                unit = CssSizeLengthUnit.NoUnit;
+                return true;
+            }
+
+            if (!IsNumberOrDot(trimmedValue[0]))
+            {
+                unit = CssSizeLengthUnit.Calc;
+                return true;
+            }
```

Comment:
Is the `unit` result backwards?

Reply:
What does `backwards` mean? The logic should be the same as before.

```csharp
            if (value.StartsWith("calc", StringComparison.OrdinalIgnoreCase))
            {
                _stringValue = value;
                _value = null;
                _unit = CssSizeLengthUnit.NoUnit;
                return;
            }

            var index = value
                .Select((c, i) => ((char c, int i)?)(c, i))
                .FirstOrDefault(x => !(x.Value.c == '.' || x.Value.c >= '0' && x.Value.c <= '9'))
                ?.i
                ?? value.Length;

            if (index == 0) // The first char is not a number or dot. Or the value is an empty string.
            {
                _stringValue = value;
                _value = null;
                _unit = CssSizeLengthUnit.Calc;
                _parsed = false;
                return;
            }
```

Reply:
I see, but we may be wrong before. If `trimmedValue` starts with `calc`, the unit should be `Calc`, and shoud be `NoUnit` while it not starts with number or dot.

Reply:
> but we may be wrong before

🤣 Really? Please make sure about it and if so, I'll make a commit for this. Also, we need to check if it's a lack of related unit tests which did not expose the issue before.

Reply:
The changes are related to [your commit](https://github.com/ant-design-blazor/ant-design-blazor/commit/c8b52156a051bb10e17c4719262bf6524cb03913#diff-0a92621cbff66fd618c8848c2cd86ad991bb9a4b20cddaab1b9ca0fabfa2ee06) @ElderJames  .

Reply:
Logically, that's true. 😅

Reply:
Fixed.